internal/clientio/iohandler/netconn/netconn.go

```diff
@@ -0,0 +1,185 @@
+package netconn
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net"
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/dicedb/dice/internal/clientio/iohandler"
+)
+
+const (
+	maxRequestSize = 512 * 1024 // 512 KB
+	readBufferSize = 4 * 1024   // 4 KB
+	idleTimeout    = 10 * time.Minute
+)
+
+var (
+	ErrRequestTooLarge = errors.New("request too large")
+	ErrIdleTimeout     = errors.New("connection idle timeout")
+	ErrorClosed        = errors.New("connection closed")
+)
+
+// IOHandler handles I/O operations for a network connection
+type IOHandler struct {
+	fd     int
+	file   *os.File
+	conn   net.Conn
+	reader *bufio.Reader
+	writer *bufio.Writer
+	logger *slog.Logger
+}
+
+var _ iohandler.IOHandler = (*IOHandler)(nil)
+
+// NewIOHandler creates a new IOHandler from a file descriptor
+func NewIOHandler(clientFD int, logger *slog.Logger) (*IOHandler, error) {
+	file := os.NewFile(uintptr(clientFD), "client-connection")
+	if file == nil {
+		return nil, fmt.Errorf("failed to create file from file descriptor")
+	}
+
+	// Ensure the file is closed if we exit this function with an error
+	var conn net.Conn
+	defer func() {
+		if conn == nil {
+			// Only close the file if we haven't successfully created a net.Conn
+			err := file.Close()
+			if err != nil {
+				logger.Warn("Error closing file in NewIOHandler:", slog.Any("error", err))
+			}
+		}
+	}()
+
+	var err error
+	conn, err = net.FileConn(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create net.Conn from file descriptor: %w", err)
+	}
+
+	return &IOHandler{
+		fd:     clientFD,
+		file:   file,
+		conn:   conn,
+		reader: bufio.NewReader(conn),
+		writer: bufio.NewWriter(conn),
+		logger: logger,
+	}, nil
+}
+
+func NewIOHandlerWithConn(conn net.Conn) *IOHandler {
+	return &IOHandler{
+		conn:   conn,
+		reader: bufio.NewReader(conn),
+		writer: bufio.NewWriter(conn),
+	}
+}
+
+func (h *IOHandler) FileDescriptor() int {
+	return h.fd
+}
+
+// ReadRequest reads data from the network connection
+func (h *IOHandler) Read(ctx context.Context) ([]byte, error) {
+	var data []byte
+	buf := make([]byte, readBufferSize)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return data, ctx.Err()
+		default:
+			err := h.conn.SetReadDeadline(time.Now().Add(idleTimeout))
+			if err != nil {
+				return nil, fmt.Errorf("error setting read deadline: %w", err)
+			}
+
+			n, err := h.reader.Read(buf)
+			if n > 0 {
+				data = append(data, buf[:n]...)
+			}
+			if err != nil {
+				switch {
+				case errors.Is(err, syscall.EAGAIN), errors.Is(err, syscall.EWOULDBLOCK), errors.Is(err, io.EOF):
```

Comment:
syscall.EAGAIN and syscall.EWOULDBLOCK can be grouped together in the `syscall.EAGAIN`